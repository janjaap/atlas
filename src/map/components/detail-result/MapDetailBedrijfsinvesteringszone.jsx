import React from 'react';
import PropTypes from 'prop-types';

import MapDetailResultItem from './MapDetailResultItem';
import MapDetailResultWrapper from './MapDetailResultWrapper';

const MapDetailBedrijfsinvesteringszone = ({
  panoUrl,
  bedrijfsinvesteringszone,
  onMaximize,
  onPanoPreviewClick
}) => (
  <MapDetailResultWrapper
    panoUrl={panoUrl}
    onMaximize={onMaximize}
    onPanoPreviewClick={onPanoPreviewClick}
    subTitle={bedrijfsinvesteringszone.label}
    title="Bedrijfsinvesteringszone"
  >
    <ul className="map-detail-result__list">
      <MapDetailResultItem
        label="Type"
        value={bedrijfsinvesteringszone.type}
      />
      <MapDetailResultItem
        label="Heffingsgrondslag"
        value={bedrijfsinvesteringszone.heffingsgrondslag}
      />
      <MapDetailResultItem
        label="Jaarlijkse heffing"
        value={bedrijfsinvesteringszone.heffingLabel}
      />
      <MapDetailResultItem
        label="Aantal heffingsplichtigen"
        value={bedrijfsinvesteringszone.heffingsplichtigen.toString()}
      />
    </ul>
  </MapDetailResultWrapper>
);

MapDetailBedrijfsinvesteringszone.propTypes = {
  bedrijfsinvesteringszone: PropTypes.shape({
    heffingLabel: PropTypes.string,
    heffingsgrondslag: PropTypes.string,
    heffingsplichtigen: PropTypes.number,
    label: PropTypes.string,
    type: PropTypes.string
  }).isRequired,
  panoUrl: PropTypes.string.isRequired,
  onMaximize: PropTypes.func.isRequired,
  onPanoPreviewClick: PropTypes.func.isRequired
};

export default MapDetailBedrijfsinvesteringszone;
