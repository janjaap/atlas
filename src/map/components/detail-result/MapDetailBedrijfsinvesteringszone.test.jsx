import React from 'react';
import { shallow } from 'enzyme';

import MapDetailBedrijfsinvesteringszone from './MapDetailBedrijfsinvesteringszone';

describe('MapDetailBedrijfsinvesteringszone', () => {
  it('should render everything', () => {
    const bedrijfsinvesteringszone = {
      heffingLabel: 'Bedrijfsinvesteringszone heffingLabel',
      heffingsgrondslag: 'Bedrijfsinvesteringszone heffingsgrondslag',
      heffingsplichtigen: 80,
      label: 'Bedrijfsinvesteringszone label',
      type: 'Bedrijfsinvesteringszone type'
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailBedrijfsinvesteringszone
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        bedrijfsinvesteringszone={bedrijfsinvesteringszone}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
