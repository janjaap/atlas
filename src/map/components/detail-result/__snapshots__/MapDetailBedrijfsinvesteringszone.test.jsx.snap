// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailBedrijfsinvesteringszone should render everything 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Bedrijfsinvesteringszone label"
  title="Bedrijfsinvesteringszone"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailResultItem
      label="Type"
      value="Bedrijfsinvesteringszone type"
    />
    <MapDetailResultItem
      label="Heffingsgrondslag"
      value="Bedrijfsinvesteringszone heffingsgrondslag"
    />
    <MapDetailResultItem
      label="Jaarlijkse heffing"
      value="Bedrijfsinvesteringszone heffingLabel"
    />
    <MapDetailResultItem
      label="Aantal heffingsplichtigen"
      value="80"
    />
  </ul>
</MapDetailResultWrapper>
`;
